metals/src/main/scala/scala/meta/internal/metals/ServerLivenessMonitor.scala

```diff
@@ -178,17 +134,25 @@ object ServerLivenessMonitor {
   object FirstPing extends State
   object Running extends State
 
-  class ReconnectOptions(optReconnect: Option[() => Future[Unit]]) {
-    val canReconnect = optReconnect.nonEmpty
-    val reconnect: () => Future[Unit] =
-      optReconnect.getOrElse(() => Future.successful(()))
-  }
-
-  object ReconnectOptions {
-    def empty = new ReconnectOptions(None)
-    def apply(reconnect: () => Future[Unit]) = new ReconnectOptions(
-      Some(reconnect)
-    )
-  }
+  def connectedParams(serverName: String, icons: Icons): MetalsStatusParams =
```

Comment:
Thank you for adding the build server name to the status! :+1: Another idea that could be a nice addition: when the Metals is trying to connect to the BSP (for example as a result of the user action after clicking on the status icon, or just in general while waiting for the build server to start) could we show appropriate message and icon? :slightly_smiling_face:

Reply:
I think we do show when we're connecting to the build server in metas status now. We could think about moving this to  this new bsp status but that's not something I want to think about in this PR.